tree/tree/src/TTree.cxx

```diff
@@ -5390,40 +5433,14 @@ Long64_t TTree::GetCacheAutoSize(Bool_t withDefault /* = kFALSE */ )
      cacheFactor = 0.0;
    }
 
-   Long64_t cacheSize = 0;
-
-   if (fAutoFlush < 0) {
-      cacheSize = Long64_t(-cacheFactor * fAutoFlush);
-   } else if (fAutoFlush == 0) {
-      const auto medianClusterSize = GetMedianClusterSize();
-      if (medianClusterSize > 0)
-         cacheSize = Long64_t(cacheFactor * 1.5 * medianClusterSize * GetZipBytes() / (fEntries + 1));
-      else
-         cacheSize = Long64_t(cacheFactor * 1.5 * 30000000); // use the default value of fAutoFlush
-   } else {
-      cacheSize = Long64_t(cacheFactor * 1.5 * fAutoFlush * GetZipBytes() / (fEntries + 1));
-   }
-
-   if (cacheSize >= (INT_MAX / 4)) {
-      cacheSize = INT_MAX / 4;
-   }
+   Long64_t cacheSize = calculateCacheSize(cacheFactor);
 
    if (cacheSize < 0) {
       cacheSize = 0;
    }
 
    if (cacheSize == 0 && withDefault) {
-      if (fAutoFlush < 0) {
-         cacheSize = -fAutoFlush;
-      } else if (fAutoFlush == 0) {
-         const auto medianClusterSize = GetMedianClusterSize();
-         if (medianClusterSize > 0)
-            cacheSize = Long64_t(1.5 * medianClusterSize * GetZipBytes() / (fEntries + 1));
-         else
-            cacheSize = Long64_t(cacheFactor * 1.5 * 30000000); // use the default value of fAutoFlush
```

Comment:
Using the new `calculateCacheSize`, this line would always use `1.0` for the cache factor (which may or may not be inteded).